### PR TITLE
feat: Add task component

### PR DESCRIPTION
**Pull request by mdeetman**
image

### CODE REVIEW

File: src/components/Task.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        fontWeight: 'bold'
+    },
+})(Typography);
+
+const useStyles = makeStyles({
+    root: {
+        flexGrow: 1,
+        maxWidth: 910,
+        '&:hover': {
+            backgroundColor: '#f8f8f8'
```
Comment (jasonenglish) [inline src/components/Task.js:27]: was the on hover effect not working?

File: src/components/Task.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        }
+    },
+}, { name: 'MuiCard' });
+
+
+export default function Task({ title, description, source,  onEdit }) {
```
Comment (jasonenglish) [inline src/components/Task.js:33]: we should require the source to be `oneOf` like we did with the SourceIcon component

**Comment by jasonenglish** [comment]
Failing build, looks like there's a missing dependency `"@material-ui/lab": "^4.0.0-alpha.56",` does not exist in the package.json file. you can install it by running `npm. i @material-ui/lab`

image

File: src/components/Task.js (hunk 1 above)
Comment (jasonenglish) [reply src/components/Task.js:27]: curious why we needed this?

Comment (mdeetman) [reply src/components/Task.js:27]: I just copied that from the css file in NewSuperTadaWeb

Comment (mdeetman) [reply src/components/Task.js:27]: Just checked and the hover color I set is the same as the default hover color so I will remove it

Comment (jasonenglish) [reply src/components/Task.js]: yeah, i think you can delete the maxWidth as well, as that i think comes from one of the components

Comment (jasonenglish) [reply src/components/Task.js]: maybe try it and see?

**Pull request merged**